pkg/utils/posthog.go

```diff
@@ -44,44 +46,106 @@ func (p *PosthogCliClient) Done() {
 	p.client.Close()
 }
 
-// CaptureBake gathers telemetry on git repos that are being baked
-//
-//nolint:errcheck
-func (p *PosthogCliClient) CaptureBake(urls []string) {
-	p.client.Enqueue(posthog.Capture{
-		DistinctId: "pizza-bakers",
-		Event:      "cli_user baked repo",
-		Properties: posthog.NewProperties().Set("clone_url", urls),
-	})
-}
-
 // CaptureLogin gathers telemetry on users who log into OpenSauced via the CLI
 //
 //nolint:errcheck
 func (p *PosthogCliClient) CaptureLogin(username string) {
-	p.client.Enqueue(posthog.Capture{
-		DistinctId: username,
-		Event:      "cli_user logged in",
-	})
+	if p.activated {
+		p.client.Enqueue(posthog.Capture{
+			DistinctId: username,
+			Event:      "cli_user logged in",
+		})
+	}
 }
 
 // CaptureFailedLogin gathers telemetry on failed logins via the CLI
 //
 //nolint:errcheck
 func (p *PosthogCliClient) CaptureFailedLogin() {
-	p.client.Enqueue(posthog.Capture{
-		DistinctId: "login-failures",
-		Event:      "cli_user failed log in",
-	})
+	if p.activated {
+		p.client.Enqueue(posthog.Capture{
+			DistinctId: "login-failures",
+			Event:      "cli_user failed log in",
+		})
+	}
+}
+
+//nolint:errcheck
```

Comment:
Are the `nolint:errcheck`s here because Posthog is fire and forget?

Reply:
Right: Posthog's client interface for the Enqueue function returns an error:

```
	// Queues a message to be sent by the client when the conditions for a batch
	// upload are met.
	// This is the main method you'll be using, a typical flow would look like
	// this:
	//
	//	client := posthog.New(apiKey)
	//	...
	//	client.Enqueue(posthog.Capture{ ... })
	//	...
	//	client.Close()
	//
	// The method returns an error if the message queue could not be queued, which
	// happens if the client was already closed at the time the method was
	// called or if the message was malformed.
	Enqueue(Message) error
```

And we don't really care (at least at this point) to log to the user that telemetry failed.